File: jwt/src/main/java/com/member/jwt/controller/SignupController.java
package com.member.jwt.controller;

import com.member.jwt.dto.SignupRequestDto;
import com.member.jwt.dto.SignupResponseDto;
import com.member.jwt.service.SignupService;
import jakarta.validation.Valid;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

@Controller
@ResponseBody
public class SignupController {

    private final SignupService signupService;

    public SignupController(SignupService signupService) {

        this.signupService = signupService;
    }

    @PostMapping("/signup")
    public ResponseEntity<SignupResponseDto> signupProcess(
            @Valid @ModelAttribute SignupRequestDto signupRequestDto,
            @RequestParam("profileImage") MultipartFile profileImage) {

        System.out.println(signupRequestDto.getEmail());
        Integer memberId = signupService.signupProcess(signupRequestDto, profileImage);

        return ResponseEntity.ok(new SignupResponseDto(memberId));
    }
}

File: jwtKotlin/src/main/kotlin/com/member/jwtkotlin/controller/SignupController.kt
package com.member.jwtkotlin.controller

import com.member.jwtkotlin.dto.SignupRequestDto
import com.member.jwtkotlin.dto.SignupResponseDto
import com.member.jwtkotlin.service.SignupService
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.multipart.MultipartFile

@Controller
@ResponseBody
class SignupController(private val signupService: SignupService) {
    @PostMapping("/signup")
    fun signupProcess(
        @ModelAttribute signupRequestDto: @Valid SignupRequestDto,
        @RequestParam("profileImage") profileImage: MultipartFile
    ): ResponseEntity<SignupResponseDto> {
        println(signupRequestDto.email)
        val memberId = signupService.signupProcess(signupRequestDto, profileImage)

        return ResponseEntity.ok(SignupResponseDto(memberId))
    }
}